File: src/api/schemas/evaluations.py
"""Pydantic schemas for evaluation list (watchlist) endpoints."""

from __future__ import annotations

from pydantic import BaseModel, Field


class EvalCreateRequest(BaseModel):
    card_name: str = Field(..., min_length=1, max_length=500)
    set_code: str | None = None
    collector_number: str | None = None
    liga_url: str | None = None
    source_data_json: str | None = None
    price_at_add: float | None = None
    card_id: int | None = None


class EvalEntryResponse(BaseModel):
    id: int
    card_name: str
    set_code: str | None
    collector_number: str | None
    liga_url: str | None
    price_at_add: float | None
    card_id: int | None
    image_url: str | None
    created_at: str | None


class EvalPromoteResponse(BaseModel):
    collection_entry_id: int
    card_name: str

File: frontend/src/api/evaluations.ts
import type { ApiResponse } from "../types/api";
import { apiGet, apiPost, apiDelete } from "./client";

export interface EvalEntry {
  id: number;
  card_name: string;
  set_code: string | null;
  collector_number: string | null;
  liga_url: string | null;
  price_at_add: number | null;
  card_id: number | null;
  image_url: string | null;
  created_at: string | null;
}

export interface EvalCreateBody {
  card_name: string;
  set_code?: string | null;
  collector_number?: string | null;
  liga_url?: string | null;
  source_data_json?: string | null;
  price_at_add?: number | null;
  card_id?: number | null;
}

export interface EvalPromoteResult {
  collection_entry_id: number;
  card_name: string;
}

export function fetchEvaluations(): Promise<ApiResponse<EvalEntry[]>> {
  return apiGet<EvalEntry[]>("/api/v1/evaluations");
}

export function createEvaluation(
  body: EvalCreateBody,
): Promise<ApiResponse<EvalEntry>> {
  return apiPost<EvalEntry>("/api/v1/evaluations", body);
}

export async function deleteEvaluation(id: number): Promise<void> {
  await apiDelete(`/api/v1/evaluations/${id}`);
}

export function promoteEvaluation(
  id: number,
): Promise<ApiResponse<EvalPromoteResult>> {
  return apiPost<EvalPromoteResult>(`/api/v1/evaluations/${id}/promote`, {});
}
